test/await-after.test.js

```diff
@@ -306,6 +308,77 @@ test('await after complex scenario', async (t) => {
   }
 })
 
+test('without autostart and sync/async plugin mix', async (t) => {
+  const app = {}
+  boot(app, { autostart: false })
+  t.plan(22)
+
+  let firstLoaded = false
+  let secondLoaded = false
+  let thirdLoaded = false
+  let fourthLoaded = false
+
+  app.use(first)
+  await app.after()
+  t.ok(firstLoaded, 'first is loaded')
+  t.notOk(secondLoaded, 'second is not loaded')
+  t.notOk(thirdLoaded, 'third is not loaded')
+  t.notOk(fourthLoaded, 'fourth is not loaded')
+
+  const contents = await fs.readFile(path.join(__dirname, 'fixtures', 'dummy.txt'), 'utf-8')
```

Comment:
what is your intention reading it twice?

Reply:
Just wanting to mix in a bit of async FS I/O in various contexts (e.g., both at the top-level and within a plugin) per @Eomm's request on the original PR: 

https://github.com/fastify/avvio/pull/173#discussion_r814227953

Reply:
OK - so it looks like this top-level async FS read (a late addition to the original PR) was what caused the test to pass without my changes.  I've removed the top-level read and verified that test fails as expected without my changes.

@Eomm - can you verify that the remaining FS read that occurs within the plugin registration function on L363 still satisfies [your original request to mix in some async FS I/O in the test](https://github.com/fastify/avvio/pull/173#discussion_r814227953)?